src/master-playlist-controller.js

```diff
@@ -510,6 +510,36 @@ export class MasterPlaylistController extends videojs.EventTarget {
     }
   }
 
+  /**
+   * Re-tune playback quality level for the current player
+   * conditions. This method will perform destructive actions like removing
+   * already buffered content in order to readjust the currently active
+   * playlist quickly. This is good for manual quality changes
+   *
+   * @private
+   */
+  fastQualityChange_() {
+    const media = this.selectPlaylist();
+
+    if (media === this.masterPlaylistLoader_.media()) {
+      return;
+    }
+
+    this.masterPlaylistLoader_.media(media);
+
+    // delete all buffered data to allow an immediate quality switch, then seek
+    // in place to give the browser a kick to remove any cached frames from the
+    // previous rendition
+    this.mainSegmentLoader_.resetEverything(() => {
+      // we want SegmentLoader.load() to be called only once the new playlist
+      // has finished loading. This prevents it from being called too soon.
+      this.hls_.ignoreNextSeekingEvent_ = true;
```

Comment:
With the merge of https://github.com/videojs/http-streaming/pull/161 `ignoreNextSeekingEvent_` was removed, and a new method of seeking was introduced. It's possible `seekTo` may be used directly and `ignoreNextSeekingEvent_` removed.

Reply:
@gesinger Hm just calling `seekTo` with the flag removed is problematic since the logic in `MasterPlaylistController.setCurrentTime()` results in `SegmentLoader.load()` getting called before the new playlist has finished loading, so we append another segment from the previous rendition before appending the new quality segments. Perhaps adding an another `return` in `MPC.setCurrentTime()` if the PlaylistLoader state is `'SWITCHING_MEDIA`` would be a possible solution to this? Off the top of my head I'm not sure what other consequences that could have though.

Reply:
If we can avoid looking at the PlaylistLoader state, that would be best. One thing that could be done here is use the normal `this.tech_.setCurrentTime` call directly, if we want to avoid the `MasterPlaylistController.setCurrentTime()` call, since `seekTo` is meant to add exactly that to the `setCurrentTime` calls. If we do use `this.tech_.setCurrentTime` directly though, it would be good to add a comment about why we can't go through the normal `seekTo` method (in this case to avoid resetting the loader, since it's a seek meant to kick things instead of actually performing a seek with buffer clearing).